backend/dataall/modules/dataset_sharing/services/share_managers/s3_bucket_share_manager.py

```diff
@@ -0,0 +1,445 @@
+import abc
+import json
+import logging
+from itertools import count
+
+from dataall.base.aws.iam import IAM
+from dataall.base.aws.sts import SessionHelper
+from dataall.core.environment.db.environment_models import Environment, EnvironmentGroup
+from dataall.modules.dataset_sharing.aws.kms_client import KmsClient
+from dataall.modules.dataset_sharing.aws.s3_client import S3ControlClient, S3Client
+from dataall.modules.dataset_sharing.db.share_object_models import ShareObject
+from dataall.modules.dataset_sharing.services.share_managers.share_manager_utils import ShareManagerUtils
+from dataall.modules.dataset_sharing.services.dataset_alarm_service import DatasetAlarmService
+from dataall.modules.datasets_base.db.dataset_models import Dataset, DatasetBucket
+from dataall.modules.dataset_sharing.db.share_object_repositories import ShareObjectRepository
+
+logger = logging.getLogger(__name__)
+
+DATAALL_READ_ONLY_SID = "DataAll-Bucket-ReadOnly"
+DATAALL_ALLOW_OWNER_SID = "AllowAllToAdmin"
+IAM_S3BUCKET_ROLE_POLICY = "dataall-targetDatasetS3Bucket-AccessControlPolicy"
```

Comment:
All the IAM policies are prefixed with the resource prefix of the environment. The default value is `dataall` but it could take other values. All resources created in the environment will be prefixed with this resource prefix and the permissions given to the pivot role will be limited to the resources prefixed with this particular prefix. 

Instead of hardcoding "dataall-targetDatasetS3Bucket-AccessControlPolicy" we need to define this string as "targetDatasetS3Bucket-AccessControlPolicy" and then append the policy to the prefix like: `f"{target_environment.resourcePrefix}-{IAM_S3BUCKET_ROLE_POLICY}"`

Reply:
(We have the same issue with `IAM_ACCESS_POINT_ROLE_POLICY = "targetDatasetAccessControlPolicy"` and now is tricky to update them)

Reply:
I am not sure I understood this one completely. Lets discuss this in the next office hours or offline.

Reply:
It is not crucial, we can leave it out of the PR